platforms/android/app/src/main/assets/www/js/index.js

```diff
@@ -62,43 +67,66 @@ openSonect: function() {
-                          },
+        {
+        uniqueIdentifier: "IBAN_1",
+        name: "My Bank",
+        detailDescription: "Balance: 20CHF",
+        funds: 20.0,
+        image: "Bank"
+        },
     ];
 
-    sonect.present(credentials, paymentMethods, theme,
-                   function(uniqueIdentifier, balanceCallback) {
-                   //This method should check against the bank balance and return a balance.
-                   let balance = {
-                        uniqueIdentifier: uniqueIdentifier,
-                        value: "20.00",
-                        currency: "CHF"
-                   };
-                   balanceCallback(balance);
-                   },
-                   function(uniqueIdentifier, value, currency, paymentCallback) {
-                   //This method should initiate bank payment and return a payment reference as a string.
-                   let paymentReference = {
-                        uniqueIdentifier: uniqueIdentifier,
-                        paymentReference: "PAYMENT_REFERENCE"
-                   };
-                   paymentCallback(paymentReference);
-                   },
-                   function(msg) {
-                   document
-                   .getElementById('deviceready')
-                   .querySelector('.received')
-                   .innerHTML = msg;
-                   },
-                   function(err) {
-                   document
-                   .getElementById('deviceready')
-                   .innerHTML = '<p class="event received">' + err + '</p>';
-                   })
+    sonect.init(credentials, paymentMethods, theme);
+
+    sonect.present(
+        function(uniqueIdentifier, balanceCallback) {
+            let balance = {
+                uniqueIdentifier: uniqueIdentifier,
+                value: "100.00",
+                currency: "CHF"
+            };
+            balanceCallback(balance);
+        },
+        function(uniqueIdentifier, value, currency, paymentCallback) {
+            //To process payment without leaving the SDK, follow this code path
+            // let paymentReference = {
+            //     uniqueIdentifier: uniqueIdentifier,
+            //     paymentReference: "PAYMENT_REFERENCE"
+            // };
+            // paymentCallback(paymentReference);
+
+            //Alternatively, hide the SDK, and call sonect.pay when you have obtained the payment reference
+            sonect.hide(null, null);
+            app.paymentUniqueIdentifier = uniqueIdentifier;
+        },
+        function(msg) {
+            document
+            .getElementById('deviceready')
+            .querySelector('.received')
+            .innerHTML = msg;
+        },
+        function(err) {
+            document
+            .getElementById('deviceready')
+            .innerHTML = '<p class="event received">' + err + '</p>';
+        }
+    )
+},
+
+paySonect: function() {
+    let paymentReference = {
+        value: "20.00",
```

Comment:
Notice that hardcoded values here since we don't know context from android. We just close SDK with values as a result and outer app knows about.